### PR TITLE
Body part 1

### PR DESCRIPTION
**Pull request by Banti140**

### CODE REVIEW

File: src/components/Body/MainMenuLists/handleClickHelper.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        return (current) => {
+            if (current.name === newCopiedDetail.name) {
+                current.quantity = current.quantity + 1;
+            }
+        }
+    },
+
+    isPresentInCart(detail, isAvilableInCart) {
+        return (dish) => {
+            if (detail.name === dish.name) {
```
Comment (kk-1onewolf) [inline src/components/Body/MainMenuLists/handleClickHelper.js:13]: Mutation of isAvailableInCart

File: src/components/Body/MainMenuLists/MainMenuData.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+class MainMenuData extends Component {
+
+  handleClick = () => {
+    const { name, price, description, pic_Id, dishClicked, id } = this.props;
+    return dishClicked({ name, price, description, pic_Id, id });
+  }
+
+  render() {
+    const { name, price, description, pic_Id } = this.props;
```
Comment (thekoolfunda) [inline src/components/Body/MainMenuLists/MainMenuData.js:14]: please add defaultProps and propTypes

File: src/components/Body/MainMenuLists/MainMenuData.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import React, { Component } from "react";
+import "./MainMenuData.css";
+import Button from "@material-ui/core/Button";
+// import Cart from "./Cart/Cart";
+
+class MainMenuData extends Component {
+
+  handleClick = () => {
+    const { name, price, description, pic_Id, dishClicked, id } = this.props;
+    return dishClicked({ name, price, description, pic_Id, id });
```
Comment (thekoolfunda) [inline src/components/Body/MainMenuLists/MainMenuData.js:10]: We can just pass the id here

File: src/components/Body/MainMenuLists/handleClick.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+function updateDishQuantityInCart(dishCopy) {
```
Comment (thekoolfunda) [inline src/components/Body/MainMenuLists/handleClick.js:1]: Filename
handleClick <> /helpers/mainMenuLists.general.js

**Pull request merged**